chore: add a section about lint (Spotless) hook in CONTRIBUTING (#420)
jenkins-infra/pipeline-library: CONTRIBUTING.md

An IDE or text editor of your choice, like [Vim](https://www.vim.org/), [Emacs](https://www.gnu.org/software/emacs/) or [VS Code](https://code.visualstudio.com/) for example.

### Add a `pre-push` git hook to check the lint before pushing

By adding the following file `pre-push` (without extension) into your local git repository `.git/hooks` folder, the command `mvn spotless:chek` will run before every push you'll make to check if your code is still correctly linted, and will prevent pushing malformatted code which would fail your build anyway.

```bash
#!/bin/sh

mvn spotless:check
```
If your push has been aborted, you can manually fix the reported errors, or use the following command to automatically fix them: `mvn spotless:apply`

Notes:

> - this would works for Linux and MacOS, you'll need to adapt it for Windows.

Author: ```suggestion
- this hook is totally optional.
- this hook works for Linux and MacOS, you'll need to adapt it for Windows.
```

An IDE or text editor of your choice, like [Vim](https://www.vim.org/), [Emacs](https://www.gnu.org/software/emacs/) or [VS Code](https://code.visualstudio.com/) for example.

### Add a `pre-push` git hook to check the lint before pushing

> By adding the following file `pre-push` (without extension) into your local git repository `.git/hooks` folder, the command `mvn spotless:chek` will run before every push you'll make to check if your code is still correctly linted, and will prevent pushing malformatted code which would fail your build anyway.

Author: ```suggestion
By adding the following file `pre-push` (without extension) into your local git repository `.git/hooks` folder, 
the command `mvn spotless:chek` will run before every push you'll make to check if your code is still correctly linted, 
and will prevent pushing malformatted code which would fail your build anyway.
```

> By adding the following file `pre-push` (without extension) into your local git repository `.git/hooks` folder, the command `mvn spotless:chek` will run before every push you'll make to check if your code is still correctly linted, and will prevent pushing malformatted code which would fail your build anyway.

Author: ```suggestion
By adding the following file `pre-push` (without extension) into your local git repository `.git/hooks` folder, 
the command `mvn spotless:chek` will run before every push you'll make to check if your code is still correctly linted, 
and will prevent pushing malformatted code which would fail your build anyway.
```
